tests/integration/apps/test_command_line_apps.py

```diff
@@ -130,37 +139,171 @@ def test_upsert_instruments_with_invalid_mapping(self):
+            "mapping": os.path.join(self.cur_dir, self.mapping_invalid),
+            "delimiter": None,
+            "scope": self.testscope,
+            "num_header": 0,
+            "num_footer": 0,
+            "debug": "debug",
+            "batch_size": None,
+            "dryrun": False,
+            "line_terminator": r"\n",
+            "display_response_head": True
+        }
+
         test_data_root = Path(__file__).parent.joinpath("test_data")
-        self.invalid_args["file_path"] = test_data_root.joinpath("transactions.csv")
-        responses = load_transactions(self.invalid_args)
+        invalid_args["file_path"] = test_data_root.joinpath("transactions.csv")
+        invalid_args["mapping"] = test_data_root.joinpath("mapping_trans_invalid.json")
+
+        responses = load_transactions(invalid_args)
 
         self.assertEqual(1, len(responses["transactions"]["errors"]))
         self.assertEqual(0, len(responses["transactions"]["success"]))
+
+    def test_upsert_quotes_with_valid_mapping(self):
+
+        valid_args = {
+            "file_path": os.path.join(self.cur_dir, self.valid_instruments),
+            "secrets_file": self.secrets,
+            "mapping": os.path.join(self.cur_dir, self.mapping_valid),
+            "delimiter": None,
+            "scope": self.testscope,
+            "num_header": 0,
+            "num_footer": 0,
+            "debug": "debug",
+            "batch_size": None,
+            "dryrun": False,
+            "line_terminator": r"\n",
+            "display_response_head": True
+        }
+
+        test_data_root = Path(__file__).parent.joinpath("test_data")
+        valid_args["file_path"] = test_data_root.joinpath("quotes.csv")
+        valid_args["mapping"] = test_data_root.joinpath("mapping_quote_valid.json")
+
+        responses = load_quotes(valid_args)
+
+        succ, err, failed = cocoon_printer.format_quotes_response(responses)
+
+        self.assertEqual(0, len(err))
+        self.assertEqual(0, len(failed))
+        self.assertEqual(14, len(succ))
+
+    def test_upsert_quotes_with_valid_mapping_check_scaled_values(self):
+
+        valid_args = {
+            "file_path": os.path.join(self.cur_dir, self.valid_instruments),
+            "secrets_file": self.secrets,
+            "mapping": os.path.join(self.cur_dir, self.mapping_valid),
+            "delimiter": None,
+            "scope": self.testscope,
+            "num_header": 0,
+            "num_footer": 0,
+            "debug": "debug",
+            "batch_size": None,
+            "dryrun": True,
+            "line_terminator": r"\n",
+            "display_response_head": True
+        }
+
+        test_data_root = Path(__file__).parent.joinpath("test_data")
+        valid_args["file_path"] = test_data_root.joinpath("quotes.csv")
+        valid_args["mapping"] = test_data_root.joinpath("mapping_quote_valid.json")
+
+        responses = load_quotes(valid_args)
+
+        test_values = list(responses["__adjusted_quote"])
```

Comment:
Use the result from responses instead of a dry-run and have two tests, one with scaling and one without.